tree_sitter_grep/src/plugin.rs

```diff
@@ -0,0 +1,80 @@
+use libloading::Library;
+use once_cell::sync::OnceCell;
+use std::ffi::OsStr;
+use tree_sitter::Node;
+use tree_sitter_grep_plugin_core::Filterer;
+use tree_sitter_grep_plugin_core::PluginDeclaration;
+
+use std::sync::Arc;
+
+pub struct FiltererProxy {
+    filterer: Box<dyn Filterer>,
+    _lib: Arc<Library>,
+}
+
+impl Filterer for FiltererProxy {
+    fn call(&self, node: &Node) -> bool {
+        self.filterer.call(node)
+    }
+}
+
+struct PluginRegistrar {
+    filterer: Option<FiltererProxy>,
+    lib: Arc<Library>,
+}
+
+impl PluginRegistrar {
+    pub fn new(lib: Arc<Library>) -> Self {
+        Self {
+            lib,
+            filterer: Default::default(),
+        }
+    }
+}
+
+impl tree_sitter_grep_plugin_core::PluginRegistrar for PluginRegistrar {
+    fn register_filterer(&mut self, filterer: Box<dyn Filterer>) {
+        let proxy = FiltererProxy {
+            filterer,
+            _lib: self.lib.clone(),
+        };
+        self.filterer = Some(proxy);
+    }
+}
+
+fn load_plugin(library_path: impl AsRef<OsStr>, filter_arg: Option<&str>) -> FiltererProxy {
```

Comment:
I think this should only be getting called once (across the multiple threads running the query + then filtering) if I understand `once_cell` correctly (ie all of those threads should be able to "compete" for who gets to initialize the `OnceCell` but only one of them "wins" and then the others' `.get_or_init()` calls would block until the one is done initializing)? But I didn't sanity-check that